brain-marks/Models/DeepLinks.swift

```diff
@@ -0,0 +1,16 @@
+//
+//  DeepLinks.swift
+//  brain-marks
+//
+//  Created by Susannah Skyer Gupta on 10/7/22.
+//
+
+import Foundation
+struct DeepLinks {
+  var url: URL? {
+  guard let url = URL(string: "brain-marks://addCategory") else {
+  fatalError("Failed to construct URL")
+  }
+    return url
```

Comment:
One thing, you can return a `URL` rather than `URL?` (optional URL) because if it can't create a URL object it'll just throw fatalError and crash, haha which is fine cause we're using a string, to always make a proper URL so this shouldn't happen

Reply:
Thank you so much; I will carefully read through these. I just pushed some new changes after re-reading https://www.donnywals.com/handling-deeplinks-in-ios-14-with-onopenurl/ and watching part of https://youtu.be/H-MNG-GS8J0. I think where I am stuck now is figuring out what state to reference and mutate from ContentView.swift to get the new case of CategorySheetView to show.

Reply:
> One thing, you can return a `URL` rather than `URL?` (optional URL) because if it can't create a URL object it'll just throw fatalError and crash, haha which is fine cause we're using a string, to always make a proper URL so this shouldn't happen

yeah, I ended up deciding this whole file was overkill and went with the far simpler:
```
.widgetURL(URL(string: "brainmarks://category/0"))
```

in the widget extension itself (BrainMarksCreateCategory.swift).

Reply:
> yeah, I ended up deciding this whole file was overkill and went with the far simpler:


That's fine! I think it's okay to keep, it's a great set up for adding more widgets in the future, but it's fine if you want to do it either way

Reply:
Yeah, if I do manage to crack this, I'll probably want to try https://github.com/mikaelacaron/brain-marks/issues/142 and then https://github.com/mikaelacaron/brain-marks/issues/155, so this file may reappear :)

Reply:
@SuzGupta  You can do it! Do you care if I tweet about this? someone may be able to chime in

Reply:
Please Do tweet about this and tag me; would love to try to pair with someone.